### PR TITLE
Add support for Unsetting Default Value of a column

### LINKED ISSUE #2835: Default value of a column can't be unset
## Description
Currently, **it is not possible to remove default value of a column.**

## Expected behavior
Mathesar UI should support removing default value of a column.

## To Reproduce
- Set default value of a column.
- Now, there is no way to unset the default value.

## Environment
 - OS: Ubuntu 22.10 (Kinetic Kudu)
 - Browser: Google Chrome
 - Browser Version: 112.0.5615.29

## Additional context
image

Clearing the input of the text fields **sets empty string as default**, instead of ```null```.
There is no ```unset/remove default``` button either.

## Reference Design
https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=5641-31738

### PR DESCRIPTION
**Pull request by Aritra8438**
Fixes #2835.

This pull request adds support for unsetting default value of a column.
- Here is the **[figma design](https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=5641-31738)** by @ghislaineguerin for UI changes.

## Technical details
**No default value** radio button will allow users to remove default of a column (i.e set default to `null`).

When checking `null`, `===` is used to check if the column default is `null`, instead of using:
```
if (!default) {
}
``` 
**not to capture** `""`, `0`, `undefined`, `false` **as null.**
 

## Screenshots

https://github.com/centerofci/mathesar/assets/64671908/589ba894-1346-41f8-bf42-2b10cbcc6ddd

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Aritra8438** [comment]
Hi, @rajatvijay, a friendly ping, in case you missed it.

**Comment by rajatvijay** [comment]
@Aritra8438, I was on a vacation during the last week. Will review this tomorrow.

File: mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte [hunk 1]
```diff
@@ -17,7 +19,13 @@
 
   $: initialValue = column.column.default?.value ?? column.initialInputValue;
   $: value = initialValue;
+  $: isDefaultNull = column.column.default === null;
   $: actionButtonsVisible = (() => {
+    if (isDefaultNull) {
+      if (typeof value === 'object') {
+        JSON.stringify(initialValue) !== '';
+      } else return String(initialValue) !== '';
```
Comment (rajatvijay) [inline mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte:27]: A simple check of the form: 

```typescript

if (isDefaultNull) {
    return column.column.default !== null;
}

```
should work just fine.

**Review by rajatvijay — CHANGES_REQUESTED** [review]
@Aritra8438 Looks good. Made two minor comments.

File: mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte (hunk 1 above)
Comment (Aritra8438) [reply mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte]: Yes, it's working fine.

File: mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte [hunk 2]
```diff
@@ -37,11 +45,15 @@
...
-      await columnsDataStore.patch(column.id, {
-        default: {
+      let defaultRequest = null;
+      if (!isDefaultNull) {
+        defaultRequest = {
           is_dynamic: !!column.column.default?.is_dynamic,
           value,
-        },
+        };
+      }
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte]: I have modified this.

**Comment by Aritra8438** [comment]
Thanks @rajatvijay, I have addressed those changes.

File: mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte [hunk 3]
```diff
@@ -37,11 +43,14 @@
   async function save() {
     typeChangeState = { state: 'processing' };
     try {
+      const defaultRequest = isDefaultNull
+        ? null
+        : {
+            is_dynamic: !!column.column.default?.is_dynamic,
+            value,
+          };
```
[the remark this answers is not in the document]
Comment (Aritra8438) [reply mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte:51]: Yes @rajatvijay.

**Review by rajatvijay — APPROVED** [review]
LGTM 🚀

**Pull request merged**